test/Sema/has_symbol.swift

```diff
@@ -120,15 +142,23 @@ protocol View {}
   static func buildEither<Content>(second content: Content) -> Content where Content : View { fatalError() }
 }
 
-struct Image : View {
-}
+struct Image : View {}
 
 struct MyView {
-  @ViewBuilder var body: some View {
-    if #_hasSymbol(global) { // expected-error {{#_hasSymbol is not supported in closures}}
-      Image()
-    } else {
-      Image()
-    }
+  let image = Image()
+  
+  @ViewBuilder var globalView: some View {
+    if #_hasSymbol(global) { image }
+    else { image }
+  }
+  
+  @ViewBuilder var ambiguousFuncView: some View {
+    if #_hasSymbol(ambiguousFunc) { image } // expected-error {{ambiguous use of 'ambiguousFunc()'}}
+    else { image }
+  }
+    
+  @ViewBuilder var localFuncView: some View {
+    if #_hasSymbol(localFunc) { image } // expected-warning {{global function 'localFunc()' is not a weakly linked declaration}}
```

Comment:
@xedin I tried these test cases out with `-enable-experimental-feature ResultBuilderASTTransform` and they are generally passing. The only issue is here, where this diagnostic starts to be emitted three times. The first two occurrences happen under `SyntacticElementSolutionApplication` and the third happens during `FunctionSyntacticDiagnosticWalker`.

Reply:
Resolved by storing an `Invalid` bit in `PoundHasSymbolInfo`.